examples/fuzz_example3/trdelnik-tests/src/fuzz_instructions.rs

```diff
@@ -0,0 +1,322 @@
+                    &fuzz_example3::ID,
+                )
+                .unwrap();
+
+            let escrow_pda_authority = fuzz_accounts
+                .escrow_pda_authority
+                .get_or_create_account(
+                    self.accounts.escrow_pda_authority,
+                    &[b"ESCROW_PDA_AUTHORITY"],
+                    &fuzz_example3::ID,
+                )
+                .unwrap();
+
+            let escrow_token_account = fuzz_accounts
+                .escrow_token_account
+                .get_or_create_account(
+                    self.accounts.escrow_token_account,
+                    client,
+                    mint,
+                    escrow_pda_authority.pubkey(),
+                    u64::MAX,
+                    None,
+                    None,
+                    0,
+                    None,
+                )
+                .unwrap();
+
+            let acc_meta = fuzz_example3::accounts::WithdrawUnlocked {
+                recipient: recipient.pubkey(),
+                recipient_token_account,
+                escrow: escrow.pubkey(),
+                escrow_token_account,
+                escrow_pda_authority: escrow_pda_authority.pubkey(),
+                mint,
+                token_program: anchor_spl::token::ID,
+                system_program: SYSTEM_PROGRAM_ID,
+            }
+            .to_account_metas(None);
+            Ok((vec![recipient], acc_meta))
+        }
+        fn check(
+            &self,
+            pre_ix: Self::IxSnapshot,
+            post_ix: Self::IxSnapshot,
+            _ix_data: Self::IxData,
+        ) -> Result<(), &'static str> {
+            if let Some(escrow) = pre_ix.escrow {
+                let recipient = pre_ix.recipient.unwrap();
+                if let Some(recepient_token_account_pre) = pre_ix.recipient_token_account {
+                    if let Some(recepient_token_account_post) = post_ix.recipient_token_account {
+                        if escrow.recipient == *recipient.key {
+                            if recepient_token_account_pre.amount
+                                == recepient_token_account_post.amount
+                            {
+                                // INFO Recipient was not able to withdraw
+                                return Err("Recipient was not able to withdraw any funds");
+                            } else if recepient_token_account_pre.amount + escrow.amount
+                                != recepient_token_account_post.amount
+                            {
+                                if recepient_token_account_pre.amount + escrow.amount
+                                    > recepient_token_account_post.amount
+                                {
+                                    // INFO The recipient was able to withdraw,
+                                    // but not as much as was initially intended.
+                                    eprintln!(
```

Comment:
the error reporting has to be improved, but before we have something better, it would make sense to return Err() for each variant rather than writing to error output...